pageserver/src/walingest.rs

```diff
@@ -330,8 +332,13 @@ impl WalIngest {
                         < 0
                     {
                         self.checkpoint.oldestXid = xlog_checkpoint.oldestXid;
-                        self.checkpoint_modified = true;
                     }
+
+                    // Write a new checkpoint key-value pair on every checkpoint record, even
+                    // if nothing really changed. Not strictly required, but it seems nice to
+                    // have some trace of the checkpoint records in the layer files at the same
+                    // LSNs.
+                    self.checkpoint_modified = true;
```

Comment:
I'd like to have some eyeballs on this change. I think this is a good idea and should not cause any harm, but am I missing some unforeseen consequences?

This is kind of a similar issue as https://github.com/neondatabase/neon/issues/5962. With this, a checkpoint record is no longer a no-op as far as the pageserver is concerned.

Reply:
@arssher, @problame

Reply:
Looks good, it may somewhat improve situation with #5962 , but can't harm anything AFAIS.